aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1689,5 +1690,89 @@ Tensor chain_matmul(TensorList matrices) {
   }
 }
 
+DEFINE_DISPATCH(unpack_pivots_stub);
+
+std::tuple<Tensor, Tensor, Tensor> _lu_unpack(
+    const Tensor& LU_data,
+    const Tensor& LU_pivots,
+    bool pivots_as_permutation_matrix
+    ) {
+  // In the generalized LU factorization, the following shape relations hold:
```

Comment:
I think we could add a couple tests here to make sure that all the shapes match properly. 
And also to make sure that `LU_pivots` is an int32 type.

Reply:
The original `lu_unpack` does not check shapes. And it is stated in the doc that it accepts input coming from `troch.lu`. Shall we extend it to arbitrary inputs? Then the question of tensor layouts becomes relevant.

Reply:
I don't think you want to extend that. But since you do reinterpret casts and a lot of assumption on the sizes, I think it's nice to have checks because users don't always read the doc ;) 
It should blow up in a nice way if they do bad things.

Reply:
OK, then I will throw a couple of tests on dtypes and the memory alignment, since I assume that `pivots` is contiguous.